common/lib/xmodule/xmodule/js/spec/problem/edit_spec_hint.coffee

```diff
@@ -0,0 +1,772 @@
+# This file tests the parsing of  extended-hints, double bracket sections {{ .. }}
+# for all sorts of markdown.
+describe 'drop down optionresponse components', ->
+  it 'multiple multiline (new-style) drop down with hints', ->
+    data = MarkdownEditingDescriptor.markdownToXml("""
+      Translation between Dropdown and ________ is straightforward.
+
+      [[
+         (Multiple Choice) 	 {{ Good Job::Yes, multiple choice is the right answer. }}
+         Text Input	                  {{ No, text input problems don't present options. }}
+         Numerical Input	 {{ No, numerical input problems don't present options. }}
+      ]]
+
+
+
+      Clowns have funny _________ to make people laugh.
+      
+      [[
+        dogs		{{ NOPE::Not dogs, not cats, not toads }}
+        (FACES)	{{ With lots of makeup, doncha know?}}
+            
+        money       {{ Clowns don't have any money, of course }}
+        donkeys     {{don't be an ass.}}
+        -no hint-
+      ]]
+
+    """)
+    expect(data).toEqual("""
+    <problem>
+    <p>Translation between Dropdown and ________ is straightforward.</p>
+    <optionresponse>
+      <optioninput>
+        <option correct="True">Multiple Choice <optionhint label="Good Job">Yes, multiple choice is the right answer.</optionhint></option>
+        <option correct="False">Text Input <optionhint>No, text input problems don't present options.</optionhint></option>
+        <option correct="False">Numerical Input <optionhint>No, numerical input problems don't present options.</optionhint></option>
+      </optioninput>
+    </optionresponse>
+
+    <p>Clowns have funny _________ to make people laugh.</p>
+    <optionresponse>
+      <optioninput>
+        <option correct="False">dogs <optionhint label="NOPE">Not dogs, not cats, not toads</optionhint></option>
+        <option correct="True">FACES <optionhint>With lots of makeup, doncha know?</optionhint></option>
+        <option correct="False">money <optionhint>Clowns don't have any money, of course</optionhint></option>
+        <option correct="False">donkeys <optionhint>don't be an ass.</optionhint></option>
+        <option correct="False">-no hint-</option>
+      </optioninput>
+    </optionresponse>
+    
+    </problem>
+    """)
+
+  it 'drop down with demand hint', ->
```

Comment:
Nitpick: can you name your tests in a way that works with the Jasmine convention? That is, `describe` is followed by a noun that you are testing the behavior of, and `it` is followed by a phrase that describes what happens in the test. For this specific case, I think a good name would be
```coffee
describe 'Markdown to XML parsing', ->
  it 'correctly parses dropdowns', ->
```